### PR TITLE
Add Celu Op

### PR DESCRIPTION
**Pull request by zrphercule**
Celu(x) = max(x, 0) + min(0, alpha * (exp(x/alpha) - 1))
See #1121 for related paper.

### CODE REVIEW

**Comment by zrphercule** [comment]
I will update the docs later for not making so many conflicts.

**Comment by ArmenAg** [comment]
This is more of a philosophical question, but at what point should we prefer Functions to adding new Op's? It seems like it's going to become tougher and tougher for back-ends to be ONNX compliant if we have an ever-growing addition of ops.

**Comment by houseroad** [comment]
According to the doc https://github.com/onnx/onnx/blob/master/docs/AddNewOp.md, if CELU can be added as simple combination, we should use function, such as LE => LT + EQ.

**Comment by ArmenAg** [comment]
So in this case CELU can be implemented using max/min/exp/mul/sub. So should it be a Function?

**Comment by houseroad** [comment]
@ArmenAg in general it depends, if it is (relatively) complicated, and in deep learning frameworks we should implement it as a separate operator. For Celu, personally, I prefer to add it as an operator, not function... because the decomposition is not straightforward. Also, for functions, I think it should not conflict with operator, actually, it can be optional attribute as part of the operator schema, just like the operator attribute.

**Comment by zrphercule** [comment]
@gramalingam @linkerzhang Any comments? Thanks!

**Comment by zrphercule** [comment]
@ArmenAg @spandantiwari @linkerzhang  How do you like if we should merge this op? Thanks!

File: onnx/defs/math/defs.cc [hunk 1]
```diff
@@ -1059,6 +1059,29 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+            AttributeProto::FLOAT,
+            1.0f)
+        .SetDoc(Celu_ver9_doc)
+        .Input(0, "input", "Input tensor", "T")
+        .Output(0, "output", "Output tensor", "T")
+        .TypeConstraint(
+            "T",
+            {"tensor(float16)", "tensor(float)", "tensor(double)"},
+            "Constrain input and output types to float tensors.")
+        .TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput));
```
Comment (spandantiwari) [inline onnx/defs/math/defs.cc]: It will be nice to have a shape inference test also.

File: docs/Changelog.md [hunk 2]
```diff
@@ -8481,6 +8481,43 @@ This version of the operator has been available since version 9 of the default O
...
 </dl>
 
+### <a name="Celu-9"></a>**Celu-9**</a>
+
+  Calculate the Celu of input tensor element-wise.
+  Celu(x) = max(x, 0) + min(0, alpha * (exp(x/alpha) - 1)).
+
+#### Version
+
+This version of the operator has been available since version 9 of the default ONNX operator set.
```
Comment (faxu) [inline docs/Changelog.md]: Is this for opset10? Opset9 released with ONNX 1.4

**Comment by faxu** [comment]
@zrphercule : is this op still needed? ONNX 1.5 will release soon - are you targeting to include this as part of opset10?

**Comment by zrphercule** [comment]
@faxu Thanks for your review. I think it is still needed, I will update it to version 10 next week with the operator LE and GE. Sounds good?

**Comment by faxu** [comment]
@zr

> @faxu Thanks for your review. I think it is still needed, I will update it to version 10 next week with the operator LE and GE. Sounds good?

ONNX 1.5 release branch is planned to be cut by the end of today. If you can only get to it next week, it'll be released as part of opset 11 in ONNX 1.6 most likely.

**Comment by zrphercule** [comment]
> @zr
> 
> > @faxu Thanks for your review. I think it is still needed, I will update it to version 10 next week with the operator LE and GE. Sounds good?
> 
> ONNX 1.5 release branch is planned to be cut by the end of today. If you can only get to it next week, it'll be released as part of opset 11 in ONNX 1.6 most likely.

Then I am okay with it to leave it to next version. Thanks!

**Comment by faxu** [comment]
@zrphercule is this change still needed? If so, can you please sign the CLA so it can be included in the 1.6 release?

**Comment by Data-drone** [comment]
was this added in the end?

**Comment by zrphercule** [comment]
> was this added in the end?

Not really since this pr is out-dated. Feel free to add it into modern onnx ;)

**Pull request closed without merging**